src/main/java/soconnect/model/SoConnect.java

```diff
@@ -132,6 +133,22 @@ public boolean hasPerson(Person person) {
         return persons.contains(person);
     }
 
+    /**
+     * Returns true if all the tags in the contact exist in the tagList.
+     *
+     * @param person The person to be added.
+     * @return True if all the tags exists in the tagList. False if otherwise.
+     */
+    public boolean areTagsAvailable(Person person) {
+        List<Tag> personTags = new ArrayList<>(person.getTags());
+        for (int i = 0; i < personTags.size(); i++) {
+            if (!tags.hasTag(personTags.get(i))) {
+                return false;
+            }
+        }
+        return true;
```

Comment:
I haven't read fully yet about the implementation of `tags`, but can we do `containsAll` here? In case the underlying implementation of `tags` is based on List.

Reply:
Oh. I didn't notice that. I tried and it works fine. Thanks for the recommendation.